File: src/main/java/com/floorestimatepro/model/PlanPoint.java
package com.floorestimatepro.model;


//A record is a special class: it stores fields and gives you accessors named:
//x()  and y()  automatically (not getX needed)

public record PlanPoint(double x, double y){
    public double distanceTo(PlanPoint other){
        double dx = this.x() - other.x();
        double dy = this.y() - other.y();
        return Math.sqrt(dx * dx + dy *dy);
    }
}

File: python/src/floor_estimate_pro/model/plan_point.py
from __future__ import annotations

from dataclasses import dataclass
import math


@dataclass(frozen=True)
class PlanPoint:
    x: float
    y: float

    def distance_to(self, other: PlanPoint) -> float:
        dx = self.x - other.x
        dy = self.y - other.y
        return math.sqrt(dx * dx + dy * dy)
